src/core_init_atmosphere/mpas_geotile_manager.F

```diff
@@ -0,0 +1,467 @@
+   !> to add that tile. If the tile is found, a pointer is returned in tile.
+   !
+   !-----------------------------------------------------------------------
+   subroutine mpas_get_tile(lat, lon, tile, tile_hash, geo_pool)
+
+      implicit none
+
+      ! Input variables 
+      real (kind=RKIND), value :: lat, lon
+      type (tile_t), intent(out), pointer :: tile
+      type (hash_table), pointer, dimension(:,:), intent(inout) :: tile_hash
+      type (mpas_pool_type), intent(inout), pointer :: geo_pool
+
+      tile => null()
+      
+      ! Search for the tile in the data structure (ds) we have implemented, if
+      ! we find the tile in the ds, then the tile_t type will be returned in 
+      ! tile.
+      !
+      ! If we haven't found the tile in the ds, then we need to add the tile
+      ! via add tile, which will also give us the tile_t type that contains
+      ! that tile.
+
+      if (mpas_search_tile(lat, lon, tile, tile_hash, geo_pool)) then
+         ! pass
+      else
+         call mpas_add_tile(lat, lon, tile, tile_hash, geo_pool)
+      endif
+
+   end subroutine mpas_get_tile
+
+   !***********************************************************************
+   !
+   !  function mpas_search_tile
+   !
+   !> \brief   Searches for an allocated datafile tile
+   !> \author  Miles A. Curry
+   !> \date    
+   !> \details 
+   !>  Given a latitude and longitude, this function searches the implemented
+   !> data structure to determine if the tile containing that latitude and
+   !> longitude have been allocated or not. A pointer to the tile_t type 
+   !> containing that tile is returned in the tile argument.
+   !>
+   !
+   !-----------------------------------------------------------------------
+   function mpas_search_tile(lat, lon, tile, tile_hash, geo_pool)
+      implicit none
+
+      ! Input variables
+      real (kind=RKIND), value :: lat, lon
+      type (tile_t), intent(out), pointer :: tile
+      type (hash_table), pointer, dimension(:,:), intent(inout) :: tile_hash
+      type (mpas_pool_type), intent(inout), pointer :: geo_pool
+      
+      ! Return value
+      logical :: mpas_search_tile
+
+      ! Local variables
+      character (len=StrKIND) :: fname
+      integer :: x, y
+      integer, pointer :: tile_nx
+      integer, pointer :: tile_ny
+
+      tile => null()
+
+      call mpas_pool_get_config(geo_pool, 'tile_x', tile_nx)
+      call mpas_pool_get_config(geo_pool, 'tile_y', tile_ny)
+
+      ! Calculate the tiles filename, and its start x and y values. We will
+      ! use these values to calculate its location in our hash table
+      call mpas_gen_file(lat, lon, fname, x, y, geo_pool)
+
+      mpas_search_tile = .FALSE.
+
+      if (associated(tile_hash( (x-1)/tile_nx , (y-1)/tile_ny ) % ptr)) then
+         tile => tile_hash( (x-1)/tile_nx , (y-1)/tile_ny ) % ptr
+         mpas_search_tile = .TRUE.
+         return
+      endif
```

Comment:
It may eliminate the need to consistently maintain a non-trivial argument list to `tile_hash` by rewriting this as:
```
      tile => tile_hash( (x-1)/tile_nx , (y-1)/tile_ny ) % ptr
      if (associated(tile)) then
         mpas_search_tile = .TRUE.
      endif
```
The alternate return can also be omitted if we choose.

Reply:
I like that change as well, I'll put these changes in. Thanks for the recommendation!